### ISSUE TITLE
Create a System Resource for pages data

### ISSUE THREAD

**Issue by kof** [webstudio-is/webstudio#4115]
As a builder I want to be able to create a navigation based on pages data without manually creating navigation items for each page.

- [ ] Provide System resource with Pages Resource
- [ ] Pages provides a nested structure for pages inside folders (User can use nested collections)
- [ ] Data provides a full url user can use in a link

**Comment by iocouto** [comment]
@kof is the idea behind this so that we'd get a site-wide resource with metadata on the site's page hierarchy — page URLs and titles — that we could use to build customised navigation items such as menus and breadcrumbs?...

**Comment by kof** [comment]
Yes, basically expose the page data via resource inside the builder.

**Comment by iocouto** [comment]
@kof ICYMI: https://discord.com/channels/955905230107738152/1410849061245947964

Is there any possibility to bribe the team so we can get this sooner rather than later?... 🙏😬
